Add C++ function for torch.distributed.tensor._op_schema.is_view_op

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #162337
* #162336
* #161695
* #162298
* #162220
* #162219
* #162218
* #161693
* #161692
* #161634
* #161633
* #161596
* __->__ #161595
* #161591
* #161590
* #161586
* #161466

This seems to have been an especially slow one because of the repeated pybind access (schema is a pybind, as is arguments, and then we hit each argument). It's still ~~1% of total benchmark runtime because of the repeated single pybind function call, but that's a lot better.

Differential Revision: [D81530095](https://our.internmc.facebook.com/intern/diff/D81530095)

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta @ezyang @msaroufim @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

swolchok committed:
RFC: Add C++ function for torch.distributed.tensor._op_schema.is_view_op

This seems to have been an especially slow one because of the repeated pybind access (schema is a pybind, as is arguments, and then we hit each argument). It's still ~~1% of total benchmark runtime because of the repeated single pybind function call, but that's a lot better.

[ghstack-poisoned]

ezyang reviewed (approved):
The name is a bit hokey (intuitively, we are just trying to test if it's a "view" op but I wouldn't block merging on that

swolchok committed:
Update on "RFC: Add C++ function for torch.distributed.tensor._op_schema.is_view_op"


This seems to have been an especially slow one because of the repeated pybind access (schema is a pybind, as is arguments, and then we hit each argument). It's still ~~1% of total benchmark runtime because of the repeated single pybind function call, but that's a lot better.

[ghstack-poisoned]

swolchok committed:
rebase on "RFC: Add C++ function for torch.distributed.tensor._op_schema.is_view_op"


This seems to have been an especially slow one because of the repeated pybind access (schema is a pybind, as is arguments, and then we hit each argument). It's still ~~1% of total benchmark runtime because of the repeated single pybind function call, but that's a lot better.

[ghstack-poisoned]

swolchok committed:
Update on "RFC: Add C++ function for torch.distributed.tensor._op_schema.is_view_op"


This seems to have been an especially slow one because of the repeated pybind access (schema is a pybind, as is arguments, and then we hit each argument). It's still ~~1% of total benchmark runtime because of the repeated single pybind function call, but that's a lot better.

[ghstack-poisoned]

swolchok commented:
@swolchok has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D81530095).

swolchok commented:
FWIW the pybind cost here still seems way too high. Need to figure out if there's a way to get around this, such as attaching _is_view_op either to FunctionSchema or to torch._C with the pure CPython API

swolchok committed:
Update on "Add C++ function for torch.distributed.tensor._op_schema.is_view_op"


This seems to have been an especially slow one because of the repeated pybind access (schema is a pybind, as is arguments, and then we hit each argument). It's still ~~1% of total benchmark runtime because of the repeated single pybind function call, but that's a lot better.

Differential Revision: [D81530095](https://our.internmc.facebook.com/intern/diff/D81530095)

cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta EikanWang jgong5 wenzhe-nrv sanchitintel

[ghstack-poisoned]

swolchok committed:
Update on "Add C++ function for torch.distributed.tensor._op_schema.is_view_op"


This seems to have been an especially slow one because of the repeated pybind access (schema is a pybind, as is arguments, and then we hit each argument). It's still ~~1% of total benchmark runtime because of the repeated single pybind function call, but that's a lot better.

Differential Revision: [D81530095](https://our.internmc.facebook.com/intern/diff/D81530095)

cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta EikanWang jgong5 wenzhe-nrv sanchitintel

[ghstack-poisoned]

swolchok commented:
@pytorchbot merge
